pkg/validator/cluster.go

```diff
@@ -434,6 +443,74 @@ func validateGeneralClusterChanges(
 	return valErrors
 }
 
+// https://github.com/bluek8s/kubedirector/issues/591
+// validateAppSpecChanges checks for modifications to app spec properties
+// at the live cluster upgrade period. There shouldn't be any difference
+// between the actual and proposed application specs, except the role image tags,
+// Version and Upgradable fields (because the target app.spec.Upgradable may have `false` value)
+// and role cardinality. Meanwhile, the cardinality changes shouldn't conflict with the
+// current cluster role members count. Otherwise, user should previously change
+// cluster role members count at a preceded cluster updating.
+func validateAppSpecChanges(
+	prevCr *kdv1.KubeDirectorCluster,
+	prevApp *kdv1.KubeDirectorApp,
+	targetApp *kdv1.KubeDirectorApp,
+) *string {
+
+	for _, appRole := range targetApp.Spec.NodeRoles {
+		// First, check that role image tags match
```

Comment:
I guess this isn't really a "check" here, but more like making sure we ignore any difference in images.

Reply:
Changed